### DISCUSSION TITLE
Roles and Permissions not read from Database on Ubuntu Linux?

### DISCUSSION THREAD

**Discussion by worgor** [spatie/laravel-permission#2283] [Q&A]
Hi, i'm using Permissions in Laravel Apps running on a Windows Apache which was working great - i've used this commands 
    $permissions = $user->getPermissionsViaRoles();
    $allpermissions = $user->getAllPermissions();
    $userroles = $user->getRoleNames()

for checking if the permissions and roles are loaded correctly, on Windows this works perfectly, on ubuntu 20.x these calls are not gettings any results and are empty. 

i've emptied the cache, actually all caches, tried to upgrade to a newer version etc. but nothing really worked - database access workes fine since i can load data without auth.

Is there anything i could do or try to overcome this problem?

Thank you very much!

**Comment by parallels999** [comment]
Make a demo app on a fresh laravel instalation
https://spatie.be/docs/laravel-permission/v5/basic-usage/new-app

**Comment by pcboii** [comment]
Also appear to be having this issue, develop/testing works fine on a MacOS environment, but when pushed to staging/production servers, no roles/permissions are applied are applied to users.

--

CAN
- Create role/permission.

CANNOT
- Apply role/permission.

--

Ubuntu 20.04 (LTS) x64
Database Connection: MySQL
spatie/laravel-permission -> 5.7.0

--

Local environment, successfully applies role to user, and can fetch user permissions.
https://app.screencast.com/jBPvgpaZQrzZq

--

Staging/Prod. on Ubuntu does not return any results.
https://app.screencast.com/QIcn6vIF4TzYz

**Reply by parallels999** [reply]
Did you clear spatie cache?

**Reply by pcboii** [reply]
Ran the following

php artisan cache:clear
php artisan config:cache
php artisan cache:forget spatie.permission.cache

**Reply by erikn69** [reply]
php artisan permission:cache-reset
